### PR TITLE
Player docs: Control Stick Input and Movement

### PR DESCRIPTION
**Pull request by engineer124**
This imports research/docs from OoT's player that is near identical to MM's, and keeps them in sync. See the relevant information pertaining to these docs here:
https://github.com/zeldaret/oot/pull/1539

Main difference is MM is still prefixing local data with `Player`, for example I went with `sPlayerControlStickMagnitude` instead of `sControlStickMagnitude`, but that's the only difference.

### CODE REVIEW

**Comment by fig02** [comment]
was wondering about that, any reason for the prefix?
I dont see any reason to when its very clear that its for player and in his overlay

**Comment by engineer124** [comment]
I have no preference on my end, I'm fine removing the prefix. This was the styling choice by @AngheloAlf and @EllipticEllipsis when they did player

**Review by fig02 — APPROVED** [review]
This approval is very biased cause much of this is my work 😄

File: src/overlays/actors/ovl_player_actor/z_player.c [hunk 1]
```diff
@@ -501,10 +501,10 @@ typedef struct struct_8085D200 {
     /* 0x9 */ u8 unk_9;
 } struct_8085D200; // size = 0xC
 
-f32 D_80862AFC; // distance of the analog stick to its center
-s16 D_80862B00; // analog stick angle/yaw
-s16 D_80862B02; // analog stick yaw + camera yaw
-s32 D_80862B04; // boolean, set to the return value of func_8083216C
+f32 sPlayerControlStickMagnitude; // distance of the analog stick to its center
+s16 sPlayerControlStickAngle;     // analog stick angle/yaw
```
Comment (fig02) [inline src/overlays/actors/ovl_player_actor/z_player.c:505]: comments can prob go actually?
I think magnitude makes more sense overall than calling it a distance. And im not sure the angle is really a "yaw", it only becomes yaw after its applied to links angle in 3d space

**Review by fig02 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**